src/utils/deploy/hash-fns.mjs

```diff
@@ -65,7 +67,12 @@ const getFunctionZips = async ({
 
   const { zipFunctions } = await import('@netlify/zip-it-and-ship-it')
 
-  return await zipFunctions(directories, tmpDir, { basePath: rootDir, config: functionsConfig })
+  return await zipFunctions(directories, tmpDir, {
+    featureFlags: { project_deploy_configuration_api_use_per_function_configuration_files: true },
```

Comment:
Another note for future reference: in the same way we have a single place for [all edge functions feature flags](https://github.com/netlify/cli/blob/63d415b4220c0988f23507ec966240754a04d59b/src/lib/edge-functions/consts.mjs#L8), it would be good to do the same for serverless functions. This will ensure the different entry points into zip-it-and-ship-it use the same set of flags, and thus avoiding a situation where two commands have different behaviour leading to a harder time troubleshooting problems.